tests/test_node.py

```diff
@@ -0,0 +1,15 @@
+import pytest
```

Comment:
This is excellent. Test cases for real-world strings should also be included here I think. E.g. like `$NODEID+12` (random example) or some such.

Reply:
I added these tests to allow testing of the methods I introduced to remove the use of eval() which as you suggest above should be part of another PR. This is a very very basic expression parser which allows addition and subtraction. The cases the functions support are based on the existing variable names in the maps.py. i.e.:

`"name": "Restore Manufacturer Defined Default Parameters %d[(sub - 3)]"` (line 184, maps.py)